gui/guiComponent.py

```diff
@@ -0,0 +1,229 @@
+import os
+from tkinter import *
+import screeninfo
+import time
+from utils.config import USE_GUI, TIMEOUT_GUI
+
+# We do not need this on Linux.
+if os.name == "nt":
+    import pythoncom
+    import win32com.client
+    import win32gui
+
+
+def windowEnumerationHandler(hwnd, top_windows):
+    """
+    Handler for Windows OS enumeration of all open windows.
+    Used to return to the Path of Exile window after displaying the overlay.
+    """
+    top_windows.append((hwnd, win32gui.GetWindowText(hwnd)))
+
+
+def windowToFront(root):
+    # This is necessary for displaying the GUI window above active window(s) on the Windows OS
+    if os.name == "nt":
+        # In order to prevent SetForegroundWindow from erroring, we must satisfy the requirements
+        # listed here:
+        # https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setforegroundwindow
+        # We satisfy this by internally sending the alt character so that Windows believes we are
+        # an active window.
+        # We need this pythoncom call for win32com use in a thread.
+        pythoncom.CoInitialize()
+        shell = win32com.client.Dispatch("WScript.Shell")
+        shell.SendKeys("%")
+        win32gui.SetForegroundWindow(root.winfo_id())
+
+
+def windowRefocus(name):
+    """
+    Restore focus to a window, if on Windows.
+    TODO: If originating window was NOT "name", return to previous window.
+    """
+
+    if os.name == "nt":
+        results = []
+        top_windows = []
+        win32gui.EnumWindows(windowEnumerationHandler, top_windows)
+        for i in top_windows:
+            if name == i[1].lower():
+                win32gui.ShowWindow(i[0], 5)
+                win32gui.SetForegroundWindow(i[0])
+                break
+
+tk = None
```

Comment:
its called only once, what is the point of even storing it when you do not reference it anywhere

Reply:
As a safety measure? but i dont know if its nessesary